### PR TITLE
Add block.shopify_attributes to blocks

### LINKED ISSUE #51: Blocks are missing {{block.shopify_attributes}}
**Issue**
All theme blocks are missing {{block.shopify_attributes}}, which prompts a warning in the Console.

**Context**
Those are still needed. It tells the editor which DOM element belongs to each block, which it uses to trigger DOM events asking you to put blocks into view when they get selected in the sidebar. I think there's a few bits of JS like this we will need to make Dawn fully functional in the theme editor.

We may need to change the DOM structure of some of the blocks, too, since I don't know that all of them have a single wrapping element - some are probably just free flowing nodes, where this feels like it requires them to be wrapped for it to make sense.

As a result, some of the axiomatic CSS might have issues i.e. .parent > * + * so we'll need to spot check that as well.

**Steps to reproduce**
Add or interact with a block in the Announcement bar section (or any section) 
Observe warning messages in the Console
https://screenshot.click/20-58-00lmy-x5i6n.gif

### PR DESCRIPTION
**Pull request by chrisberthe**
**Why are these changes introduced?**

Fixes #51.

**What approach did you take?**

Added `block.shopify_attributes` to each block inside of `section.blocks`. More information: https://shopify.dev/api/liquid/objects/block#block-shopify_attributes.

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=120813486102)
- [Editor](https://os2-demo.myshopify.com/admin/themes/120813486102/editor)

**Checklist**
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: sections/main-product.liquid [hunk 1]
```diff
@@ -69,235 +69,237 @@
...
-            </div>
-            {%- form 'product', product, id: 'product-form-installment', class: 'installment caption-large' -%}
-              <input type="hidden" name="id" value="{{ product.selected_or_first_available_variant.id }}">
-              {{ form | payment_terms }}
-            {%- endform -%}
-          {%- when 'description' -%}
-            {%- if product.description != blank -%}
-              <div class="product__description rte">
-                {{ product.description }}
+          <div {{ block.shopify_attributes }}>
```
[the remark this answers is not in the document]
Comment (mmorissette) [reply sections/main-product.liquid]: Yup, the platform automatically add the attributes to the app block wrapper.

File: sections/apps.liquid [hunk 2]
```diff
@@ -1,6 +1,8 @@
 <div class="{% if section.settings.include_margins %}page-width{% endif %}">
   {%- for block in section.blocks -%}
-    {% render block %}
+    <div {{ block.shopify_attributes }}>
```
[the remark this answers is not in the document]
Comment (mmorissette) [reply sections/apps.liquid]: Yup, the platform automatically add the attributes to the app block wrapper.

File: sections/main-product.liquid [hunk 3]
```diff
@@ -73,15 +73,15 @@
...
               {{- block.settings.text -}}
             </p>
           {%- when 'title' -%}
-            <h1 class="product__title">
+            <h1 class="product__title" {{ block.shopify_attributes }}>
               {{ product.title | escape }}
             </h1>
           {%- when 'price' -%}
-            <div class="no-js-hidden" id="price-{{ section.id }}">
+            <div class="no-js-hidden" id="price-{{ section.id }}" {{ block.shopify_attributes }}>
```
Comment (owain) [inline sections/main-product.liquid:84]: This doesn't wrap the form, but maybe that's ok?

File: sections/main-product.liquid [hunk 4]
```diff
@@ -252,7 +252,7 @@
               </div>
             </noscript>
           {%- when 'buy_buttons' -%}
-            <product-form class="product-form">
+            <product-form class="product-form" {{ block.shopify_attributes }}>
```
Comment (owain) [inline sections/main-product.liquid:255]: This misses `<pickup-availability>` but maybe ok?

File: sections/newsletter.liquid [hunk 5]
```diff
@@ -6,12 +6,12 @@
...
         {%- when 'heading' -%}
-          <h2 class="h1">{{ block.settings.heading | escape }}</h2>
+          <h2 class="h1" {{ block.shopify_attributes }}>{{ block.settings.heading | escape }}</h2>
         {%- when 'paragraph' -%}
-          <div class="newsletter__subheading rte">{{ block.settings.paragraph }}</div>
+          <div class="newsletter__subheading rte" {{ block.shopify_attributes }}>{{ block.settings.paragraph }}</div>
         {%- when 'email_form' -%}
           {% form 'customer', class: 'newsletter-form' %}
-            <input type="hidden" name="contact[tags]" value="newsletter">
+            <input type="hidden" name="contact[tags]" value="newsletter" {{ block.shopify_attributes }}>
```
Comment (owain) [inline sections/newsletter.liquid:14]: Should this all be wrapped?

File: sections/main-product.liquid (hunk 3 above)
Comment (chrisberthe) [reply sections/main-product.liquid:84]: It's works as expected but let's add a wrapper to remain consistent with other blocks.

File: sections/main-product.liquid (hunk 4 above)
Comment (chrisberthe) [reply sections/main-product.liquid]: https://github.com/Shopify/dawn/pull/68#discussion_r660959217

**Review by Thibaut — APPROVED** [review]

**Pull request merged**